### PR TITLE
Manager: fix message when title of story ends with "/" and render all components in story that were correctly created.

### LINKED ISSUE #26642: [Bug]: accidentally ending a title with a `/` breaks storybook
### Describe the bug

with a title like:

```ts
  title: "scenes/MessageComposition/shared/ComposerImageLibraryPanel/",
```

Storybook fails to compile

### To Reproduce

create a story with this title:
```ts
  title: "scenes/MessageComposition/shared/ComposerImageLibraryPanel/",
```

### System

```bash
Storybook Environment Info:

  System:
    OS: macOS 14.4
    CPU: (10) arm64 Apple M1 Pro
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.11.1 - ~/.nvm/versions/node/v20.11.1/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v20.11.1/bin/yarn
    npm: 10.2.4 - ~/.nvm/versions/node/v20.11.1/bin/npm <----- active
  Browsers:
    Chrome: 123.0.6312.58
    Safari: 17.4
```

### Additional context

image

### PR DESCRIPTION
**Pull request by hmcostaa**
Closes #26642 

## What I did

When creating a story with a title that ended with a "/", the explorer would catch that error and display the error with a message hard to parse.
Now, instead of displaying the message: "Invalid part '${name}', leading to id === parentId ('${id}'), inside title '${title}'", we have a better message that clarifies the error: "Invalid title "${title}" ending in slash: ${id}" and additionaly all other components created correctly will be displayed along side the error message.

#### The changes in this PR are covered in the following automated tests:
- [x] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

**Comment by valentinpalkovic** [comment]
Hi @hmcostaa 

Thank you so much for opening this PR.

Could you please look at whether this is a duplicate of https://github.com/storybookjs/storybook/pull/26711?

Much appreciated!

**Comment by hmcostaa** [comment]
Hi @valentinpalkovic!

Thanks for the comment. #26711 implemented a solution that is not inline with the desired behaviour for an error. Basically It's solution was that whenever a title for a story ends with a "/" it will remove it so we don't have an error.
What I did was to keep the error but clarify the message of the error.
Previously was: "Invalid part '${name}', leading to id === parentId ('${id}'), inside title '${title}",
Now is: Invalid title "${title}" ending in slash: ${id}

I also made the explorer render all other components that were correctly created below the error message and only the story with the error will not be rendered.

I'm hoping to have this fix merged so if you need extra clarification or want some change to be made, I will answer as quickly as I can.

Thanks!

File: code/ui/manager/src/components/sidebar/Refs.tsx [hunk 1]
```diff
@@ -145,7 +169,23 @@ export const Ref: FC<RefType & RefProps & { status?: State['status'] }> = React.
...
+                  isBrowsing={isBrowsing}
+                  isMain={isMain}
+                  refId={refId}
+                  data={filteredIndex}
+                  docsMode={docsOptions.docsMode}
+                  selectedStoryId={selectedStoryId}
+                  onSelectStoryId={onSelectStoryId}
+                  highlightedRef={highlightedRef}
+                  setHighlightedItemId={setHighlightedItemId}
+                />
```
Comment (ndelangen) [inline code/ui/manager/src/components/sidebar/Refs.tsx:186]: If there's an error, it's possible that trying to render the Tree will fail, right?

File: code/lib/manager-api/src/lib/stories.ts [hunk 2]
```diff
@@ -196,9 +196,7 @@ export const transformStoryIndexToStoriesHash = (
       if (parent === id) {
         throw new Error(
           dedent`
-          Invalid part '${name}', leading to id === parentId ('${id}'), inside title '${title}'
-          
-          Did you create a path that uses the separator char accidentally, such as 'Vue <docs/>' where '/' is a separator char? See https://github.com/storybookjs/storybook/issues/6128
```
Comment (ndelangen) [inline code/lib/manager-api/src/lib/stories.ts:201]: I like the solution proposed here:
https://github.com/storybookjs/storybook/pull/26911/files

WDYT? @hmcostaa

Comment (hmcostaa) [reply code/lib/manager-api/src/lib/stories.ts:201]: That was one possible solution. I just thought of an additional feature that would complement this problem by rendering the correctly typed stories.
